### PR TITLE
addresses issue #1075

### PR DESCRIPTION
**Pull request by Baalmart**
**_WHAT DOES THIS PR DO?_**
- Closes https://github.com/airqo-platform/AirQo-api/issues/1075

**_WHAT ISSUES ARE RELATED TO THIS PR?_**

- Jira cards
    - [AN-170]
- GitHub issues
    - Closes https://github.com/airqo-platform/AirQo-api/issues/1075

**_HOW DO I TEST OUT THIS PR?_**
```
cd src/auth-service
npm run stage-mac
```

🚨👇 Please ensure that your `.env` has values for the following variable: 🚨👇

- _MONGO_STAGE_URI_

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 

- [x] [request access](https://airqo-engineering.postman.co/workspace/internal~7a8e9fad-2d13-4d27-af27-f1f4e9a114bf/request/12513372-053764bb-d840-4cf8-88d2-ffbb8a347247)

[AN-170]: https://airqoteam.atlassian.net/browse/AN-170?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/auth-service/utils/email.msgs.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ module.exports = {
       "we shall get back to you as soon as possible. \n\n" +
       "Before utilising the AirQo data, your application record has to undergo " +
       "the process of approval by the AirQo platform administration.\n" +
-      "Once your application is reviewed, you will receive an email on the mailbox \n\n" +
+      "Once your application is approved, you will receive a confirmation email\n\n" +
```
Comment (Baalmart) [inline src/auth-service/utils/email.msgs.js:24]: The email template message changed as requested.

File: src/auth-service/services/mailer.js [hunk 2]
```diff
@@ -17,6 +17,8 @@ const mailer = {
 
       const mailOptions = {
         from: constants.EMAIL,
+        name: "AirQo Data Team",
```
Comment (Baalmart) [inline src/auth-service/services/mailer.js:20]: adding a sender name to the email in addition to making configurations in Gmail (See attached screenshot)
Unfortunately, this is not bringing the desired change we need., Anyway, let it be in as we continue to investigate. 
Screenshot 2022-09-25 at 18 52 21

File: src/auth-service/utils/multitenancy.js [hunk 3]
```diff
@@ -1,29 +1,27 @@
 const mongodb = require("../config/dbConnection");
 const constants = require("../config/constants");
-const { logElement, logText, logObject } = require("../utils/log");
 
 /****
  * creating a new mongoDB connection by switching tenant
  * using this to create a new connection based on tenant ID
  */
-const getTenantDB = (tenantId, modelName, schema) => {
+function getTenantDB(tenantId, modelName, schema) {
```
Comment (Baalmart) [inline src/auth-service/utils/multitenancy.js:8]: Addressing a potential runtime error relation to TYPES

File: src/auth-service/utils/request.js [hunk 4]
```diff
@@ -106,9 +106,7 @@ const request = {
               data: createdCandidate,
               status,
             });
-          }
-
-          if (responseFromSendEmail.success === false) {
+          } else if (responseFromSendEmail.success === false) {
```
Comment (Baalmart) [inline src/auth-service/utils/request.js:109]: addressing a potential runtime error and sending HEADERS after they are already sent.

File: src/auth-service/services/mailer.js (hunk 2 above)
Comment (FaithDaka) [reply src/auth-service/services/mailer.js:20]: Thanks for this @Baalmart. 

I think the desired change was to have a professional name for the _data@airqo.net_ email responses. I can see that has been changed. Was there supposed to be another outcome?

Otherwise I see a need to have the **name** attribute in the mail options as a variable, so that it's changed once rather than manually changing it everywhere it appears in the file.

**Review by 123MwanjeMike — CHANGES_REQUESTED** [review]
Thanks @Baalmart.
A few comments however.

File: src/auth-service/config/constants.js [hunk 5]
```diff
@@ -16,7 +16,7 @@ const prodConfig = {
 };
 
 const stageConfig = {
-  MONGO_URI: process.env.MONGO_GCE_URI,
+  MONGO_URI: process.env.MONGO_STAGE_URI,
```
[the remark this answers is not in the document]
Comment (Baalmart) [reply src/auth-service/config/constants.js:19]: I already notice this and took care of it by updating the secrets accordingly. @123MwanjeMike 

The challenge with the current approach is that it was hard to differentiate staging and production so it took me a long time to realise that we were using the same variable for staging and production while giving them different values!
The current setup kinda creates a maintenance nightmare.

Happy to hear your additional thoughts.

File: src/auth-service/services/mailer.js (hunk 2 above)
Comment (Baalmart) [reply src/auth-service/services/mailer.js:20]: Thanks @FaithDaka for this. I guess we can definitely make it an ENV. Shall update PR asap.

As for the change, appears it is not "working" after many tests. There is some background magic that could be beyond our control!

Perhaps we just need to do a little more research on this. 🤔

File: src/auth-service/services/mailer.js [hunk 6]
```diff
@@ -17,6 +17,8 @@ const mailer = {
 
       const mailOptions = {
         from: constants.EMAIL,
+        name: constants.EMAIL_NAME,
```
Comment (Baalmart) [inline src/auth-service/services/mailer.js:20]: @FaithDaka 
The name is now a constant variable. I couldn't use ENV because of the spaces in the name.

Otherwise, please note that the emails still come in as "data" despite all our efforts! 😁😁

Happy to hear additional thoughts/feedback.

Thanks!

File: src/auth-service/config/constants.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (123MwanjeMike) [reply src/auth-service/config/constants.js:19]: Hi @Baalmart 
The concern I have with it is the redundancy the approach creates.
It also means we are having production secrets added to the staging .env which is security wise not advisable.

[the remark this answers is not in the document]
Comment (Baalmart) [reply src/auth-service/config/constants.js:19]: @123MwanjeMike 
Perhaps we should not have the production one added to the staging ENV. 
Perhaps you have just not realised the maintenance nightmare this current approach is.

Imagine, as a developer, having to update the value of the ENV variable when testing out different environment. Instead of just running different commands.

It makes development more cumbersome and yet the shared security concerns are some debatable.

Happy to jump on a call to showcase the additional complexities the current approach brings.

In summary, what the current approach attempts to do is already taken care of in the config file of the microservice(s). The ENVS are supposed to be different variables for reasons already shared above.

File: src/auth-service/services/mailer.js (hunk 6 above)
Comment (Baalmart) [reply src/auth-service/services/mailer.js]: Hi @FaithDaka , this has now been modified as per the learnings from another PR: https://github.com/airqo-platform/AirQo-api/pull/1119

You can now test out and share more feedback accordingly.

Thanks!

File: src/auth-service/config/constants.js [hunk 7]
```diff
@@ -1,13 +1,13 @@
...
+  MONGO_URI: process.env.MONGO_DEV_URI,
   DB_NAME: process.env.MONGO_DEV,
   PWD_RESET: `${process.env.PLATFORM_DEV_BASE_URL}/reset`,
   LOGIN_PAGE: `${process.env.PLATFORM_DEV_BASE_URL}/login`,
   FORGOT_PAGE: `${process.env.PLATFORM_DEV_BASE_URL}/forgot`,
   PLATFORM_BASE_URL: process.env.PLATFORM_DEV_BASE_URL,
 };
 const prodConfig = {
-  MONGO_URI: process.env.MONGO_GCE_URI,
+  MONGO_URI: process.env.MONGO_PROD_URI,
```
[the remark this answers is not in the document]
Comment (Baalmart) [reply src/auth-service/config/constants.js:10]: Hi @123MwanjeMike , all the logic for having one variable is handled inside the configuration files of the microservices.

The ones in the ENV files are supposed to be named differently.

Perhaps we need to have a proper onboarding for some of these microservices as we continue to refactor them.

Thanks!

[the remark this answers is not in the document]
Comment (Baalmart) [reply src/auth-service/config/constants.js:10]: **Additional notes:**
In the future, whenever we make these changes, we will want to take them through a PR review process. Just like what we are doing here in this PR. We should not discover such changes by chance. @123MwanjeMike

**Review by 123MwanjeMike — APPROVED** [review]

File: src/auth-service/config/constants.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (123MwanjeMike) [reply src/auth-service/config/constants.js:19]: Thanks @Baalmart 
I think I now understand the exhaustion this creates during development.

Although, what I was trying to break is the dependency of the codebase to the environment. In that we(would) always have update the code whenever we have a new environment for the api to run in. In other words we would have 5 different environment variables for just the database connection if we had 5 separate environments instead of just the 2 (stage and prod) we currently have.

But we can for now proceed with it and refactor it in the future.
